### ISSUE TITLE
Bug: Tables lose their sorting status when interacting with them

### ISSUE THREAD

**Issue by thecreazy** [buildo/bento-design-system#262]
### Description of the bug

Each time a rerendering of the data is performed (change filter or for example set a column as favorite) the table lose the information about the sorting status, for example:

external interaction:
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/4126645/178263351-fffdd68d-f339-49a4-829c-0958a12e7400.gif)

internal interaction
![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/4126645/178263589-041dcea8-0643-472f-98b4-cdf7e14f3d85.gif)

I know that's probably a `react-table` issue but maybe, given that the code is something like
```
 <Table
        columns={columns}
        data={data}
        customSorting={tableCustomSorting}
        initialSorting={[{ id: "traffic", desc: true }]}
      />  
```

using rerendering, if we were given props like: `onSortChanged` which returns the sort key, we could pass as new `initialSorting` key the key just received

### Version

0.11.3

### What browsers are you seeing the problem on?

Chrome, Safari, Firefox, Edge

**Comment by gabro** [comment]
Hi @thecreazy, thanks for reporting!

In your example, is `customSorting` memoized (with `useCallback` for instance)? This is required by the underlying react-table API, but I've just realized we don't advertise this in the documentation.

**Comment by thecreazy** [comment]
@gabro  yes this is the actual implementation of the customSorting



```ts
  const tableCustomSorting: TableCustomSorting = useCallback((rows, sortFns) => {
    const otherSortingFns = <T extends typeof rows[number]>(a: T, b: T) => {
      let res = 0;
      for (let i = 0; i < sortFns.length; i++) {
        res = sortFns[i](a, b);
        if (res !== 0) {
          break;
        }
      }

      return res;
    };

    return rows.slice().sort((a, b) => {
      const res = starSorting(a.original.star?.value, b.original.star?.value);
      if (res === 0) return otherSortingFns(a, b);

      return res;
    });
  }, []);
```

**Comment by gabro** [comment]
Hi @thecreazy, we found the bug and fixed it. It was a default behavior of react-table that we flipped. It will be released as 0.12.1 soon

**Comment by thecreazy** [comment]
thx @gabro
